src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/String/KeyAnalyzer.cs

```diff
@@ -296,5 +318,11 @@ private sealed class JustifiedCaseInsensitiveSubstringComparer : SubstringCompar
             public override bool Equals(string? x, string? y) => x.AsSpan(IsLeft ? Index : (x!.Length + Index), Count).Equals(y.AsSpan(IsLeft ? Index : (y!.Length + Index), Count), StringComparison.OrdinalIgnoreCase);
             public override int GetHashCode(string s) => Hashing.GetHashCodeOrdinalIgnoreCase(s.AsSpan(IsLeft ? Index : (s.Length + Index), Count));
         }
+
+        private sealed class JustifiedCaseInsensitiveAsciiSubstringComparer : SubstringComparer
+        {
+            public override bool Equals(string? x, string? y) => x.AsSpan(IsLeft ? Index : (x!.Length + Index), Count).Equals(y.AsSpan(IsLeft ? Index : (y!.Length + Index), Count), StringComparison.OrdinalIgnoreCase);
+            public override int GetHashCode(string s) => Hashing.GetHashCodeOrdinalIgnoreCaseAscii(s.AsSpan(IsLeft ? Index : (s.Length + Index), Count));
```

Comment:
is this the win here?  that hashing can go through this fast path?

Reply:
That's the win for .NET Framework. The GetHashCodeOrdinalIgnoreCase method on .NET Framework calls MemoryExtensions.ToUpper(sourceSpan, destinationSpan), but on .NET Framework that's implemented effectively with ToString().ToUpper() (on .NET Core it's much better than that, but then the second change in this PR also removes that code entirely for .NET Core, which is part of where the .NET Core boost comes from).